packages/carbon-web-components/src/components/side-panel/side-panel-header.ts

```diff
@@ -0,0 +1,276 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2019, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { classMap } from 'lit/directives/class-map.js';
+import { LitElement, html } from 'lit';
+import {
+  property,
+  queryAssignedElements,
+  queryAssignedNodes,
+  state,
+} from 'lit/decorators.js';
+import { prefix } from '../../globals/settings';
+import { carbonElement as customElement } from '../../globals/decorators/carbon-element';
+import '../button/index';
+import ArrowLeft16 from '@carbon/icons/lib/arrow--left/16';
+import Close20 from '@carbon/icons/lib/close/20';
+
+import styles from './side-panel.scss';
+
+/**
+ * SidePanel header.
+ *
+ * @element cds-side-panel-header
+ */
+@customElement(`${prefix}-side-panel-header`)
+class CDSSidePanelHeader extends LitElement {
+  private _handleNavigateBack(triggeredBy: EventTarget | null) {
+    this.dispatchEvent(
+      new CustomEvent(
+        (this.constructor as typeof CDSSidePanelHeader).eventNavigateBack,
+        {
+          composed: true,
+          detail: {
+            triggeredBy,
+          },
+        }
+      )
+    );
+  }
+
+  private _handleClose(triggeredBy: EventTarget | null) {
+    this.dispatchEvent(
+      new CustomEvent(
+        (this.constructor as typeof CDSSidePanelHeader).eventClose,
+        {
+          composed: true,
+          detail: {
+            triggeredBy,
+          },
+        }
+      )
+    );
+  }
+
+  /**
+   * Determines if the title will animate on scroll
+   */
+  @property({ reflect: true, attribute: 'animate-title', type: Boolean })
+  animateTitle = true;
+
+  /**
+   * Sets the close button icon description
+   */
+  @property({ reflect: true, attribute: 'close-icon-description' })
+  closeIconDescription = 'Close';
+
+  /**
+   * Sets the current step of the side panel
+   */
+  @property({ reflect: true, attribute: 'current-step', type: Number })
+  currentStep;
+
+  /**
+   * Sets the label text which will display above the title text
+   */
+  @property({ reflect: true, attribute: 'label-text' })
+  labelText;
+
+  /**
+   * Sets the icon description for the navigation back icon button
+   */
+  @property({ reflect: true, attribute: 'navigation-back-icon-description' })
+  navigationBackIconDescription = 'Back';
+
+  // /**
+  //  * Provide a `Slug` component to be rendered inside the `SidePanel` component
+  //  */
+  // @property({ reflect: true })
+  // slug;
+
+  /**
+   * Sets the subtitle text
+   */
+  @property({ reflect: true })
```

Comment:
should add types for all properties

Reply:
> should add types for all properties
Done, although not sure what to do with Enum, will have a look around.